packages/vite/src/node/plugins/css.ts

```diff
@@ -142,19 +142,19 @@ export type ResolvedCSSOptions = Omit<CSSOptions, 'lightningcss'> & {
 export function resolveCSSOptions(
   options: CSSOptions | undefined,
 ): ResolvedCSSOptions | undefined {
-  if (options?.lightningcss) {
+  if (options?.transformer === 'lightningcss') {
     return {
       ...options,
       lightningcss: {
         ...options.lightningcss,
         targets:
-          options.lightningcss.targets ??
+          options.lightningcss?.targets ??
           convertTargets(ESBUILD_MODULES_TARGET),
       },
     }
   }
   // TS doesn't narrow the type with the previous if :/
-  return options as Omit<CSSOptions, 'lightningcss'>
+  return options as Omit<CSSOptions, 'lightningcss'> | undefined
```

Comment:
Could we change it like this so that we don't lie the types? (The change in line 145 made it possible to have `lightning.css.target` to be `undefined` when `options.transformer !== 'lightningcss'`)
```suggestion
  return { ...options, lightningcss: undefined }
```